tests/functional/context/types/test_event.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+
+from vyper.context.types.meta.event import Event
+from vyper.utils import keccak256
+
+EVENT_ID_TESTS = [
+    ("event MyLog: pass", "MyLog()"),
+    (
+        """event MyLog:
+    arg1: indexed(Bytes[3])""",
+        "MyLog(bytes)",
+    ),
+    (
+        """event MyLog:
+    arg1: indexed(int128)
+    arg2: indexed(bool)
+    arg3: indexed(address)""",
+        "MyLog(int128,bool,address)",
+    ),
+    (
+        """event Potato:
+    arg1: indexed(Bytes[3])
+    arg2: Bytes[4]
+    arg3: indexed(address)
+    arg4: indexed(int128)
+    """,
+        "Potato(bytes,bytes,address,int128)",
+    ),
+    (
+        """event Sausage:
+    arg1: int128[2]
+    arg2: uint256[3]
+    arg3: int128[2][2]""",
+        "Sausage(int128[2],uint256[3],int128[2][2])",
+    ),
+    (
+        """event MyLog:
+    arg1: Bytes[4]
+    arg2: indexed(String[100])
+    arg3: Bytes[33]""",
+        "MyLog(bytes,string,bytes)",
+    ),
+    (
+        """event FooBar:
+    arg1: int128
+    arg2: Bytes[4]
+    arg3: Bytes[39912]
+    arg4: address
+    arg5: address
+    arg6: uint256[3]""",
+        "FooBar(int128,bytes,bytes,address,address,uint256[3])",
+    ),
+    (
+        """event Nonsense:
+    a: Bytes[666]
+    b: int128[4][38]
+    c: indexed(address[3])
+    d: uint256
+    e: String[1]""",
+        "Nonsense(bytes,int128[4][38],address[3],uint256,string)",
+    ),
+]
+
+
+@pytest.mark.parametrize("source,signature", EVENT_ID_TESTS)
+def test_event_id(build_node, source, signature):
+    node = build_node(source)
+    event = Event.from_EventDef(node)
+
+    event_id = int(keccak256(signature.encode()).hex(), 16)
```

Comment:
Usually with these types of tests, it's better to hardcode the actual expected hash, to show you are getting what you expect to get.